src/commands/source/download.rs

```diff
@@ -277,39 +297,69 @@ pub async fn download(
                     }
 
                     if source_path_exists && !force {
-                        Err(anyhow!("Source exists: {}", source.path().display()))
+                        info!("Source already exists: {}", source.path().display());
+                        Ok(DownloadResult::Skipped)
                     } else {
+                        let mut download_result: Option<DownloadResult> = None;
```

Comment:
This approach isn't nice and we should try to avoid it - especially with the `unwrap()` below. Ideally you can use delayed initialization here (see https://doc.rust-lang.org/nomicon/checked-uninit.html for an example) - otherwise Rust can't save us from panics due to programming mistakes.